src/System.IO.Compression/src/System/IO/Compression/GZipStream.cs

```diff
@@ -142,7 +142,7 @@ protected override void Dispose(bool disposing)
                 {
                     _deflateStream.Dispose();
                 }
-                _deflateStream = null;
+                _deflateStream = null!;
```

Comment:
This seems like is not needed anymore.

Reply:
Oh sorry, Line 160 confused me, but actually it is marked as non-nullable, I think it should be nullable instead and add some asserts.

Reply:
Its always being newly initiated with contructor, nulled only within Dispose which is completely fine. As CheckDeflateStream() doing null check no need extra assert

Reply:
I mean you should make `_deflateStream` nullable, just as I mentioned here: https://github.com/dotnet/corefx/pull/41322#discussion_r328738155 and https://github.com/dotnet/corefx/pull/41322#discussion_r328737844